nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ConsumeEWS.java

```diff
@@ -343,7 +354,7 @@ protected void fillMessageQueueIfNecessary(ProcessContext context) throws Proces
 
             try {
                 //Get Folder
-                Folder folder = getFolder(service);
+                Folder folder = getFolder(service, context);
```

Comment:
I'd suggest moving `String MailboxProperty = context.getProperty(MAILBOX).getValue();` up here, and passing in the MAILBOX as an optional parameter to `getFolder`. Then you can keep `getFolder` as an Exchange only method.